ts/features/zendesk/screens/ZendeskAskPermissions.tsx

```diff
@@ -350,6 +373,23 @@ const ZendeskAskPermissions = () => {
     />
   );
 
+  if (
+    getZendeskTokenStatus === ZendeskTokenStatusEnum.REQUEST &&
+    isUserLoggedIn
+  ) {
+    return <LoadingSpinnerOverlay isLoading />;
+  }
+
+  if (
+    getZendeskTokenStatus === ZendeskTokenStatusEnum.ERROR &&
+    isUserLoggedIn
+  ) {
+    navigation.navigate(ZENDESK_ROUTES.MAIN, {
+      screen: ZENDESK_ROUTES.ERROR_REQUEST_ZENDESK_TOKEN
+    });
+    return undefined;
```

Comment:
Even if the navigation is executed first, it might be safer to return `null` rather than `undefined`, as the latter is not a valid `JSX` element. What do you think?

Reply:
Using `return null` generates a console warning, so it would be preferable to use `return undefined` or `return;`